scipy/spatial/transform/rotation.py

```diff
@@ -872,3 +874,31 @@ def __getitem__(self, indexer):
                   array.
         """
         return self.__class__(self._quat[indexer], normalized=True)
+
+    @classmethod
+    def random(cls, num=None, state=None):
+        """Generate rotations uniformly distributed in 3D space.
+
+        Parameters
+        ----------
+        num : None or int
+            Number of random rotations to generate. If `None`, then a single
+            rotation is generated. Default is `None`.
+        state : None or `numpy.random.RandomState` object
+            Object for manually setting the generator state.
+
+        Returns
+        -------
+        rotation : `Rotation` instance
+            Contains a single rotation if `num` is `None`. Otherwise contains a
+            stack of `num` rotations.
+        """
+        if state is not None:
```

Comment:
You should handle it in a different way, definitely not setting the global state.

Check how random state is handled in this function https://github.com/scikit-learn/scikit-learn/blob/a24c8b46/sklearn/utils/validation.py#L617

Eventually you need to call `random_state.normal`.